### PR TITLE
Populate the SNP/TDX Machine State field with the verified SNP/TDX attestation data + use a stable COS image version

### PR DESCRIPTION
**Pull request by yawangwang**
This PR includes changes to populate the machinState proto with the verified SNP/TDX attestation data. The attestation verifier service will leverage the verified machineState to make claims around.

# Breaking changes:
- Unexport `verifyGceTechnology` because GCETechnology event log is closely related to TPM, this function is directly called after the event log is parsed.
- Use a stable COS image version `cos-dev-117-18374-0-0` rather than the latest COS image version to avoid CS image build issues.

### CODE REVIEW

**Review by JoshuaKrstic — APPROVED** [review]

File: server/verify.go
Comment (alexmwu) [inline server/verify.go:328]: This should go in `parsePCClientEventLog` since it depends on the resulting state. Long term, we will have branching to attempt to get a full MachineState from TeeAttestation. If successful, we shouldn't need to read TPM-related data. If unsuccessful (e.g., TDX machine without RTMR support or SNP machine right now), then we merge in TPM MachineState.

**Review by alexmwu — CHANGES_REQUESTED** [review]

File: server/verify.go
Comment (yawangwang) [reply server/verify.go]: Discussed offline, added a wrapper function around `parsePCClientEventLog` and `VerifyGceTechnology`

File: server/verify.go
Comment (alexmwu) [inline server/verify.go:383]: I liked your suggestion of `parseMachineStateFromTPM` better than `parseTCGEventLog`. Since it is also doing event log replay and event parsing. The term "parse" is a bit overloaded unfortunately.

File: server/verify.go
Comment (alexmwu) [inline server/verify.go:182]: We should avoid proto.Merge where possible as it can override existing values.

File: server/verify.go
Comment (alexmwu) [inline server/verify.go:395]: The point of the wrapper func for parsePCClientEventlog is to avoid proto.Merge. proto.Merge is less readable since it implicitly copies every field from src to dst. This requires our implementation to be mutually exclusive, or we risk overwriting. We originally used proto.Merge to merge the state from two separate event logs, but we could just populate the TeeAttestation field directly in the wrapper.

File: server/verify.go
Comment (alexmwu) [inline server/verify.go:321]: If we unexport this, it's a backwards incompatible change. Please add a breaking comment to the PR description.

File: server/verify.go
Comment (alexmwu) [reply server/verify.go:395]: IOW, this can be inline in parseTCGEventLog.

File: server/verify_test.go
Comment (alexmwu) [inline server/verify_test.go:1068]: Please add tests for the failure cases

**Review by alexmwu — CHANGES_REQUESTED** [review]

File: server/verify.go
Comment (yawangwang) [reply server/verify.go:321]: Done.

File: server/verify.go
Comment (yawangwang) [reply server/verify.go:396]: My take from "inline" is call `parseTEEAttestation` from `parseMachineStateFromTPM` after `verifyGceTechnolgy` is done. LMK if this is not what you want.

File: server/verify.go
Comment (yawangwang) [reply server/verify.go]: Done.

File: server/verify.go
Comment (yawangwang) [reply server/verify.go]: Done.

File: server/verify_test.go
Comment (yawangwang) [reply server/verify_test.go]: Added a new test `TestParseTEEAttestation` to cover both successful and failure cases.

File: launcher/cloudbuild.yaml
Comment (alexmwu) [inline launcher/cloudbuild.yaml:4]: Please do not submit until you get approval from @jkl73 for this change.

File: server/verify.go
Comment (alexmwu) [inline server/verify.go:401]: I may be missing something: don't we already do these checks in verifyGceTechnology?? Can we  have that return pb.MachineState or just remove the call to parseTEEAttestation and do
`ms.TeeAttestation = attestation.TeeAttestation`?

**Review by alexmwu — APPROVED** [review]

File: launcher/cloudbuild.yaml
Comment (jkl73) [reply launcher/cloudbuild.yaml:4]: lgtm, this is the same base image on the last dev build.

**Review by jkl73 — APPROVED** [review]

File: server/verify.go
Comment (yawangwang) [reply server/verify.go:401]: `ms.TeeAttestation = attestation.TeeAttestation` will not work because they're two different proto types.
`parseTEEAttestation` doesn't do any checks on attestation. This function, in short term, just populates the TeeAttestation field using a deep copy of the verified attestation data, but in long run, we'll want to get a full machineState from TDX/SNP attestation. The intent of having this function is for future-proof purpose.

**Pull request merged**
